nemo_aligner/data/nlp/datasets.py

```diff
@@ -429,3 +430,144 @@ def __getitem__(self, idx):
             "labels": label_tensor,
         }
         return output
+
+
+class SteerLM2Dataset(GPTSFTChatDataset):
+    def get_prompt(self, system_turn, prompt_turns):
+        prompt = f"{self.special_tokens['system_turn_start']}System{self.special_tokens['end_of_name']}"
+        prompt += f"{system_turn}{self.special_tokens['end_of_turn']}"
+        for turn in prompt_turns:
+            prompt += f"{self.special_tokens['turn_start']}{turn['from']}{self.special_tokens['end_of_name']}"
+            prompt += f"{turn['value']}{self.special_tokens['end_of_turn']}"
+        return prompt
+
+    def _process_example(self, example):
+        """
+        Create an example by concatenating text and answer.
+        Truncation is carried out when needed, but it is performed only on the prompt side.
+        BOS, EOS, and SEP, are added if specified.
+        """
+        assert len(example["prompt_turns"]) % 2 == 1, "Number of prompt turns should be odd"
+        prompt = self.get_prompt(example["system"], example["prompt_turns"])
+        batched_token_ids = []
+        batched_masks = []
+        response_from = example["responses"][0]["from"]
+        assert [item["from"] for item in example["responses"]] == [response_from] * len(
+            example["responses"]
+        ), "All responses should be from the same person"
+        prompt += f"{self.special_tokens['turn_start']}{response_from}{self.special_tokens['end_of_name']}"
+        if "label" in example and example["label"] is not None:
+            prompt += f"{self.special_tokens['label_start']}{example['label']}{self.special_tokens['end_of_turn']}"
+        prompt_tokens = self.tokenizer.text_to_ids(prompt)
+        num_prompt_tokens = len(prompt_tokens)
+        batch_size = len(example["responses"])
```

Comment:
is it possible to directly call super()._process_example and then do the weight and logP/Q thing? If we need to do this logic then how come steerlm1 didn't need this?